src/plugins/data_views/server/deprecations/scripted_fields.ts

```diff
@@ -7,89 +7,139 @@
+              ignoreTag: true,
+            }),
+            i18n.translate('dataViews.deprecations.scriptedFields.manualStepThreeMessage', {
+              defaultMessage:
+                'Alternatively, you can achieve similar functionality by computing values at query time using the Elasticsearch Query Language (ES|QL).',
+            }),
+          ],
+        },
+      },
+    ];
   },
 });
 
-export function hasScriptedField(indexPattern: IndexPatternAttributesWithFields) {
-  if (indexPattern.fields) {
+export function hasScriptedField(dataView: DataViewAttributesWithFields) {
+  if (dataView.fields) {
     try {
-      return JSON.parse(indexPattern.fields).some(
-        (field: { scripted?: boolean }) => field?.scripted
-      );
+      return JSON.parse(dataView.fields).some((field: { scripted?: boolean }) => field?.scripted);
     } catch (e) {
       return false;
     }
   } else {
     return false;
   }
 }
+
+const dataViewIdLabel = i18n.translate('dataViews.deprecations.scriptedFields.dataViewIdLabel', {
+  defaultMessage: 'ID',
+});
+
+const dataViewNameLabel = i18n.translate(
+  'dataViews.deprecations.scriptedFields.dataViewNameLabel',
+  {
+    defaultMessage: 'Name',
+  }
+);
+
+const dataViewSpacesLabel = i18n.translate(
+  'dataViews.deprecations.scriptedFields.dataViewSpacesLabel',
+  {
+    defaultMessage: 'Spaces',
+  }
+);
+
+const buildDataViewsListEntry = (
+  so: SavedObjectsFindResult<DataViewAttributesWithFields>
+) => `- **${dataViewIdLabel}:** ${so.id}
+  - **${dataViewNameLabel}:** ${
+  so.attributes.name
+    ? `!{tooltip[${so.attributes.name}](${so.attributes.title})}`
+    : so.attributes.title
+}
+  - **${dataViewSpacesLabel}:** ${so.namespaces?.join(', ')}`;
+
+const buildMessage = ({
+  dataViewsWithScriptedFields,
+  docLinks,
+}: {
+  dataViewsWithScriptedFields: Array<SavedObjectsFindResult<DataViewAttributesWithFields>>;
+  docLinks: DocLinks;
+}): DeprecationDetailsMessage => ({
+  type: 'markdown',
+  content: i18n.translate('dataViews.deprecations.scriptedFieldsMessage', {
+    defaultMessage: `You have {numberOfDataViewsWithScriptedFields} {numberOfDataViewsWithScriptedFields, plural, one {data view} other {data views}} containing scripted fields. Scripted fields are deprecated and will be removed in the future.
+
+The ability to create new scripted fields in the Data Views management page has been disabled in 9.0, and it is recommended to migrate to [runtime fields]({runtimeFieldsLink}) or the [Elasticsearch Query Language (ES|QL)]({esqlLink}) instead.
+
+The following is a list of all data views with scripted fields and their associated spaces:
+{dataViewsList}`,
+    values: {
+      numberOfDataViewsWithScriptedFields: dataViewsWithScriptedFields.length,
+      runtimeFieldsLink: docLinks.indexPatterns.runtimeFields,
+      esqlLink: docLinks.query.queryESQL,
+      dataViewsList: dataViewsWithScriptedFields.map(buildDataViewsListEntry).join('\n'),
```

Comment:
This is looking great, tested and works as expected, no scripted fields in a data view, no tab, just the manual way using the old URL to create a scripted field, which then shows the tab with scripted fields, and no button 🥳 

One question, about the deprecation message, previously we showed just a max of 3 data views, but it seems we show them all now? or do I miss something. ... just for the case when there are many data views with scripted fields

Reply:
We used to only show up to three titles in the message, but included the full list in the corrective actions: https://github.com/elastic/kibana/blob/cdb5a2dca2f42edbe2be47c970423d7dbd1ce99e/src/plugins/data_views/server/deprecations/scripted_fields.ts#L64-L72

I removed the ones from `correctiveActions` and instead put a formatted list in the main message. This could get really long for customers who have many data views with scripted fields, but the alternative is not having access to a full list, so I'm not sure what else we could do here.

Reply:
Ah, that's how it works, thx, what else we can do? Nothing. Approving 👍